### PR TITLE
Make cache TTL mandatory

### PR DESCRIPTION
**Pull request by nwalters512**
This will ensure that Redis can make intelligent choices about when to evict values.

### CODE REVIEW

File: apps/prairielearn/src/question-servers/freeform.js [hunk 1]
```diff
@@ -2010,7 +2010,7 @@ module.exports = {
       // tl;dr: don't cache any results that would create course issues.
       const hasCourseIssues = computedData?.courseIssues?.length > 0;
       if (cacheKey && !hasCourseIssues) {
-        cache.set(cacheKey, computedData);
+        cache.set(cacheKey, computedData, config.questionRenderCacheTtlSec * 1000);
```
Comment (nwalters512) [inline apps/prairielearn/src/question-servers/freeform.js:2013]: The tradeoff here: we'd previously cache things indefinitely, assuming no memory pressure. This means that we'll recompute question panels *slightly* more often than before, which I think is ok.

File: apps/prairielearn/src/question-servers/freeform.js [hunk 2]
```diff
@@ -1984,7 +1984,7 @@ module.exports = {
     try {
       const commitHash = await courseUtil.getOrUpdateCourseCommitHashAsync(course);
       const dataHash = objectHash({ data, context }, { algorithm: 'sha1', encoding: 'base64' });
-      return `${commitHash}-${dataHash}`;
+      return `question:${commitHash}-${dataHash}`;
```
Comment (nwalters512) [inline apps/prairielearn/src/question-servers/freeform.js:1987]: Adding a scope to cache keys is generally a good idea, especially now that it's being used for multiple things.

File: apps/prairielearn/src/lib/cache.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export async function init() {
   }
 }
 
-export function set(key: string, value: any, maxAgeMS?: number) {
+export function set(key: string, value: any, maxAgeMS: number) {
```
Comment (mwest1066) [inline apps/prairielearn/src/lib/cache.ts:44]: We could remove the `?? undefined` when we use `maxAgeMS` below?

Also lol about the comment that we don't care whether it errors.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/lib/cache.ts (hunk 3 above)
Comment (nwalters512) [reply apps/prairielearn/src/lib/cache.ts:44]: 😭 I know, we should really improve that, at the very least an OTel metric (these are way too high-volume to report to Sentry).

I'll save that for the future though.

Comment (nwalters512) [reply apps/prairielearn/src/lib/cache.ts:44]: Removed `?? undefined` in 9319a08 (I misstyped "fallbacks" as "callbacks" in the commit message lol)

**Pull request merged**